extensions/amp-sticky-ad/0.1/amp-sticky-ad.js

```diff
@@ -27,14 +27,14 @@ class AmpStickyAd extends AMP.BaseElement {
   constructor(element) {
     super(element);
 
+    /** @const @private {!../../../src/service/vsync-impl.Vsync} */
+    this.vsync_ = this.getVsync();
```

Comment:
Well, here I'm moving down the path of "change less". My initial rule was: "do not override constructor ever and treat buildCallback as a constructor". This rule has been broken at some point, I assume to fix the types. I think that rule itself was probably a mistake. Not yet sure how to go forward from here. Suggestions are welcome.

As it stands, however, ampdoc-level services are not allowed in constructor. The reason is simple - the element must be in DOM to find out it's parent ampdoc.